pkg/initializer/validator/datavalidator_test.go

```diff
@@ -0,0 +1,210 @@
+			err = os.Rename(tempDir, restoreDataDir)
+			if err != nil {
+				Fail(fmt.Sprintf("Unable to rename %s to %s: %v", tempDir, restoreDataDir, err))
+			}
+		})
+	})
+
+	Context("with data directory present", func() {
+		Context("with incorrect data directory structure", func() {
+			Context("with missing member directory", func() {
+				It("should return DataDirStatus as DataDirectoryInvStruct or DataDirectoryError, and nil error", func() {
+					memberDir := path.Join(restoreDataDir, "member")
+					tempDir := fmt.Sprintf("%s.%s", memberDir, "temp")
+					err = os.Rename(memberDir, tempDir)
+					if err != nil {
+						Fail(fmt.Sprintf("Unable to rename %s to %s: %v", memberDir, tempDir, err))
+					}
+					dataDirStatus, err := validator.Validate()
+					Expect(err).ShouldNot(HaveOccurred())
+					Expect(int(dataDirStatus)).Should(SatisfyAny(Equal(DataDirectoryInvStruct), Equal(DataDirectoryError)))
+					err = os.Rename(tempDir, memberDir)
+					if err != nil {
+						Fail(fmt.Sprintf("Unable to rename %s to %s: %v", tempDir, memberDir, err))
+					}
+				})
+			})
+			Context("with member directory present", func() {
+				Context("with missing snap directory", func() {
+					It("should return DataDirStatus as DataDirectoryInvStruct or DataDirectoryError, and nil error", func() {
+						snapDir := path.Join(restoreDataDir, "member", "snap")
+						tempDir := fmt.Sprintf("%s.%s", snapDir, "temp")
+						err = os.Rename(snapDir, tempDir)
+						if err != nil {
+							Fail(fmt.Sprintf("Unable to rename %s to %s: %v", snapDir, tempDir, err))
+						}
+						dataDirStatus, err := validator.Validate()
+						Expect(err).ShouldNot(HaveOccurred())
+						Expect(int(dataDirStatus)).Should(SatisfyAny(Equal(DataDirectoryInvStruct), Equal(DataDirectoryError)))
+						err = os.Rename(tempDir, snapDir)
+						if err != nil {
+							Fail(fmt.Sprintf("Unable to rename %s to %s: %v", tempDir, snapDir, err))
+						}
+					})
+				})
+				Context("with missing wal directory", func() {
+					It("should return DataDirStatus as DataDirectoryInvStruct or DataDirectoryError, and nil error", func() {
+						walDir := path.Join(restoreDataDir, "member", "wal")
+						tempDir := fmt.Sprintf("%s.%s", walDir, "temp")
+						err = os.Rename(walDir, tempDir)
+						if err != nil {
+							Fail(fmt.Sprintf("Unable to rename %s to %s: %v", walDir, tempDir, err))
+						}
+						dataDirStatus, err := validator.Validate()
+						Expect(err).ShouldNot(HaveOccurred())
+						Expect(int(dataDirStatus)).Should(SatisfyAny(Equal(DataDirectoryInvStruct), Equal(DataDirectoryError)))
+						err = os.Rename(tempDir, walDir)
+						if err != nil {
+							Fail(fmt.Sprintf("Unable to rename %s to %s: %v", tempDir, walDir, err))
+						}
+					})
+				})
+			})
+		})
+		Context("with correct data directory structure", func() {
+			Context("with inconsistent revision numbers between etcd and latest snapshot", func() {
```

Comment:
Why do we need this test? Its just starting etcd twice right?

Reply:
This test is just to make sure the `CheckRevisionConsistency()` function at https://github.com/gardener/etcd-backup-restore/pull/85/files#diff-2140fd19fbdf869ddf7687b0f3a3bfb0R349 works as expected. And this seemed to be the most straight-forward way to simulate a new etcd with lower revision number. Any suggestions on this matter?

Reply:
I am fine with the approach, only thing this test is incorrect because of wrong calculation on revision on latest snapshot. For which i have suggested fix in one of the comments as a part of review.

In addition, in general instead of starting etcd and populating, i think it would be nice to push sample backup structure with full and delta snapshots with some predefined key in github repo itself. And use that. But i would keep that as a part of refactoring the test case related code after merging this PR.